pkg/3scale/amp/operator/system_options_provider.go

```diff
@@ -427,13 +430,19 @@ func (s *SystemOptionsProvider) setNodeAffinityAndTolerationsOptions() {
 	s.options.SphinxTolerations = s.apimanager.Spec.System.SphinxSpec.Tolerations
 }
 
-func (s *SystemOptionsProvider) setFileStorageOptions() {
+func (s *SystemOptionsProvider) setFileStorageOptions() error {
 	if s.apimanager.Spec.System != nil &&
 		s.apimanager.Spec.System.FileStorageSpec != nil &&
 		s.apimanager.Spec.System.FileStorageSpec.S3 != nil {
 		s.options.S3FileStorageOptions = &component.S3FileStorageOptions{
 			ConfigurationSecretName: s.apimanager.Spec.System.FileStorageSpec.S3.ConfigurationSecretRef.Name,
+			S3STSEnabled:            s.apimanager.Spec.System.FileStorageSpec.S3.S3STSEnabled,
 		}
+		val, err := s.isSTS()
+		if err != nil {
+			return err
+		}
+		s.options.S3FileStorageOptions.S3STSEnabled = val
```

Comment:
On second review this flow seems strange to me. We set the whether sts is enabled based on value set in apimanager cr. Then we check the secret and reset the value based on what's present in the secrets. Seems to me that the function should check if the expected keys are present and have values set and return an error. We don't need to return a bool from the function because it's already provided in the apimanager cr.

Reply:
@laurafitzgerald we discussed with @eguzki  that Operator is recognizing STS from Secret configuration

Reply:
done. Thank you